modules/apps/transfer/keeper/relay_forwarding_test.go

```diff
@@ -42,6 +42,12 @@ func (suite *KeeperTestSuite) assertAmountOnChain(chain *ibctesting.TestChain, b
 	suite.Require().Equal(amount, total.Amount, fmt.Sprintf("Chain %s: got balance of %d, wanted %d", chain.Name(), total.Amount, amount))
 }
 
+func newDenomWithAddedTrace(original types.Denom, newTrace types.Trace) types.Denom {
+	newDenom := original
+	newDenom.Trace = append([]types.Trace{newTrace}, newDenom.Trace...)
+	return newDenom
+}
+
```

Comment:
is this really that useful?

I feel like this is just a specific use case of the constructor and we can just do it directly. WDYT?

```golang
newDenom := types.NewDenom(original.Base, append([]types.Trace{newTrace}, oldDenom.Trace...))
```

Reply:
Yeah I might have been carried away with the helpers :D